client_test.go

```diff
@@ -132,8 +132,7 @@ func TestReconnecting(t *testing.T) {
 
 	prefix := "test."
 
-	client := NewStatsdClient(udpAddr.String(), prefix)
-	client.reconnect_ticker = time.NewTicker(10 * time.Millisecond)
+	client := NewStatsdClient(udpAddr.String(), prefix, AutoReconnect(10*time.Millisecond))
```

Comment:
Do we have a test to cover the specific issue we are fixing?

Reply:
There is a test that was created in #2 to test the reconnect logic. It creates a connection closes it and then tries to send data. The reconnect code reestablishes the connection and the data gets sent successfully.

Reply:
@Jonnymcc thanks for the info! I am confused. If we had a test cover this scenario, why it did not capture this issue and then we had to make this new PR? 
(@dcarney FYI)